cli/src/commands/workspace/add.rs

```diff
@@ -117,6 +120,33 @@ pub fn cmd_workspace_add(
             .unwrap()
             .to_string()
     };
+
+    if let Some(git_backend) = old_workspace_command.git_backend() {
+        let should_add_worktree = if git_backend.git_workdir().is_some() {
+            !args.no_colocate
+        } else {
+            args.colocate
+        };
```

Comment:
Nit: Can this move into a config flag? I dislike that we do it implicitly.

Reply:
as I'm thinking about the use-case here https://github.com/martinvonz/jj/pull/2512

Reply:
I was generally trying to make it follow what you wanted in that repo -- for me the decision to use collocation is at least repo-wide. E.g. if you use Nix in a repo, it needs a .git, all the time, as everything breaks apart silently if you forget. See the issue linked at the top.

All these UI choices would be much easier if you could change your mind about collocation after you had created or cloned the repo or created a workspace. It just wouldn't matter so much. I don't think it would be that hard to write.

Reply:
For the time being we can just follow `jj git init` and make it entirely dependent on the `--colocate` flag and just require people to remember to use it. Ok?

Reply:
Done, explicit --colocate only.